### PR TITLE
goimapnotify: init at 2.0

### PR DESCRIPTION
**Pull request by wohanley**
###### Motivation for this change

goimapnotify is more reliable in my experience than the alternative currently available on nixpkgs, node-imapnotify.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by puzzlewolf — CHANGES_REQUESTED** [review]
Thanks for your first PR :) :rocket: 

Please squash your commits into one commit `goimapnotify: init at 2.0`. 

The binary builds and runs fine for me. However, I don't know enough about go packaging to review this. @marsam, @doronbehar, what do you think?

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -7922,6 +7922,12 @@
     githubId = 895853;
     name = "Frank Doepper";
   };
+  wohanley = {
```
Comment (doronbehar) [inline maintainers/maintainer-list.nix:7925]: Usually there's a separate commit with:

```
maintainers: add wohanley
```

**Review by doronbehar — CHANGES_REQUESTED** [review]
Besides the comment I left, it seems to build and run but I don't get any output when I run:

```
./result-bin/bin/goimapnotify -conf ~/.config/goimapnotify.conf -list
```

No idea why's that. Also, the project seems somewhat unmaintained, at least for 5 months or so:

https://gitlab.com/shackra/goimapnotify/-/issues/4 && https://gitlab.com/shackra/goimapnotify/activity

**Comment by wohanley** [comment]
re empty output: not sure what that's about. could maybe tell you more if I saw what's in your goimapnotify.conf?

re project maintenance: not sure what's going on there either, I'm not personally involved in the project beyond packaging it here.

**Review by doronbehar — APPROVED** [review]
@wohanley I've managed to make it work after changing some variables in my config, at least for `-list`. I wouldn't say this project has the best UX regarding error handling. It also printed my boxes in a somewhat malformed format, but I guess that shouldn't hold it from getting into our package collection if you'd be willing to maintain it.

**Comment by wohanley** [comment]
@doronbehar yep, i'm happy to take care of it. thanks!

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (puzzlewolf) [reply maintainers/maintainer-list.nix:7925]: I've seen both versions on PRs. But I agree, I like the separate commit better :)

**Comment by wohanley** [comment]
Is there something else I need to do here? Sorry, this process is new to me.

**Comment by marsam** [comment]
LGTM, sorry for the delay

**Comment by wohanley** [comment]
Thanks!

**Pull request merged**